charms/kfp-viewer/src/charm.py

```diff
@@ -8,130 +8,79 @@
 """
 
 import logging
-from pathlib import Path
 
-import yaml
-from oci_image import OCIImageResource, OCIImageResourceError
+import lightkube
+from charmed_kubeflow_chisme.components.charm_reconciler import CharmReconciler
+from charmed_kubeflow_chisme.components.kubernetes_component import KubernetesComponent
+from charmed_kubeflow_chisme.components.leadership_gate_component import LeadershipGateComponent
+from charmed_kubeflow_chisme.kubernetes import create_charm_default_labels
+from lightkube.resources.apiextensions_v1 import CustomResourceDefinition
+from lightkube.resources.core_v1 import ServiceAccount
+from lightkube.resources.rbac_authorization_v1 import Role, RoleBinding
 from ops.charm import CharmBase
 from ops.main import main
-from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
 
-log = logging.getLogger()
+from components.model_name_gate_component import ModelNameGate
+from components.pebble_component import PebbleServiceContainerComponent
 
+logger = logging.getLogger(__name__)
 
-class KfpViewer(CharmBase):
-    """Charm for the Kubeflow Pipelines Viewer.
+K8S_RESOURCE_FILES = ["src/templates/auth_manifests.yaml.j2", "src/templates/crds.yaml.j2"]
 
-    https://github.com/canonical/kfp-operators
-    """
 
+class KfpViewer(CharmBase):
     def __init__(self, *args):
```

Comment:
Need to add docstring to charm and to __init__()

Reply:
Charm already has one and added to init(). Please resolve if the fixups solve the issue.